src/_json.js

```diff
@@ -20,7 +20,15 @@ function json_parse(_, val) {
   if (typeof val !== 'object' || val === null) {
     return val;
   } else if ('@ref' in val) {
-    return new values.Ref(val['@ref']);
+    var ref = val['@ref'];
+
+    if (!('class' in ref) && !('database' in ref))
+      return values.Native.fromName(ref['id']);
+
+    var cls = json_parse(_, ref['class']);
```

Comment:
I think it's preferable to call `parseJSON`, instead. I don't feel comfortable passing the `_` variable further.

Reply:
`parseJSON` receives a string not an object